activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -361,6 +361,10 @@ def test_unscoped_with_named_scope_should_not_have_default_scope
     assert_equal 10, DeveloperCalledJamis.unscoped { DeveloperCalledJamis.poor }.length
   end
 
+  def test_unscoped_with_joins_should_not_have_default_scope
+    assert_equal FirstPost.unscoped { Comment.joins(:first_post).to_a }, Comment.joins(:post).to_a
```

Comment:
I wanted to join a model with default_scope to unscope.
Because `FirstPost` has [default_scope](https://github.com/rails/rails/blob/bdeeca84e34d3e0863a6fd03defb9c95fd0241d9/activerecord/test/models/post.rb#L189) and `Post` is its unscoped version, I choosed it.

> This test is hard to follow.

I see. `:first_post` may not be a good name for this test case. I should use `:post_with_default_scope`.
I'll change so.

> Why the unscoped version of Comment.joins(:first_post).to_a is equals to Comment.joins(:post).to_a. Should not the join be removed?

That's because the unscope target is not Comment but FirstPost.